docs/decisions/0039-template-engine.md

```diff
@@ -0,0 +1,74 @@
+---
+nav_order: 39
+parent: Decision Records
+---
+
+# Template engine
+
+## Context and Problem Statement
+
+We need to choose a template engine for [custom export filters](https://docs.jabref.org/collaborative-work/export/customexports) and [AI features](https://github.com/JabRef/jabref/pull/11884).
+
+A discussion of template engines was also in [one of the JabRef repos](https://github.com/koppor/jabref/issues/392).
+
+A discussion was raised on StackOverflow ["Velocity vs. FreeMarker vs. Thymeleaf"](https://stackoverflow.com/q/1459426/10037342).
+
+## Decision Drivers
+
+* It should support different formats, in case JabRef changes in the future. Also, AI features require just plain text manipulation.
```

Comment:
```suggestion
* Text output has to be supported.
```

I don't think we work on binary output? We currently support HTML and other text output formats.

Maybe you mean

* Special escape modes for special languages (e.g., XML or JavaScript) are nice to be future proof.

(After reading the Thymeleaf link. They have a different syntax for each dialect)

But I don't think, this is important.